Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -399,13 +399,22 @@ export class BotProjectDeploy {
         spaces: 4,
       });
 
-      // Assign a LUIS key to the endpoint of each app
-      const getAccountUri = `${luisEndpoint}/luis/api/v2.0/azureaccounts`;
-      const options = {
-        headers: { Authorization: `Bearer ${this.accessToken}`, 'Ocp-Apim-Subscription-Key': luisAuthoringKey },
-      } as rp.RequestPromiseOptions;
-      const response = await rp.get(getAccountUri, options);
-      const jsonRes = JSON.parse(response);
+      let jsonRes;
+      try {
+        // Assign a LUIS key to the endpoint of each app
+        const getAccountUri = `${luisEndpoint}/luis/api/v2.0/azureaccounts`;
+        const options = {
+          headers: { Authorization: `Bearer ${this.accessToken}`, 'Ocp-Apim-Subscription-Key': luisAuthoringKey },
+        } as rp.RequestPromiseOptions;
+        const response = await rp.get(getAccountUri, options);
+        jsonRes = JSON.parse(response);
+      } catch (err) {
+        // handle the token invalid
```

Comment:
I think we should get a bit more sophisticated here, and look at the details of the error. 

IF the error is about an expired token, throw the token-specific error

Otherwise, pass back the error details.

Reply:
I check the message include token expiry or not.